pkg/scaffolding/scaffold_test.go

```diff
@@ -0,0 +1,169 @@
+//go:build !integration
+// +build !integration
+
+package scaffolding
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"knative.dev/func/pkg/filesystem"
+
+	. "knative.dev/func/pkg/testing"
+)
+
+// TestWrite_RuntimeErrors ensures that known runtimes which are not
+// yet implemented return a "not yet available" message, and unrecognized
+// runtimes state as much.
+func TestWrite_RuntimeErrors(t *testing.T) {
+	tests := []struct {
+		Runtime  string
+		Expected any
+	}{
+		{"go", nil},
+		{"python", nil},
+		{"rust", &ErrDetectorNotImplemented{}},
+		{"node", &ErrDetectorNotImplemented{}},
+		{"typescript", &ErrDetectorNotImplemented{}},
+		{"quarkus", &ErrDetectorNotImplemented{}},
+		{"java", &ErrDetectorNotImplemented{}},
+		{"other", &ErrRuntimeNotRecognized{}},
+	}
+	for _, test := range tests {
+		t.Run(test.Runtime, func(t *testing.T) {
+			// Since runtime validation during signature detection is the very first
+			// thing that occurs, we can elide most of the setup and pass zero
+			// values for source directory, output directory and invocation.
+			// This may need to be expanded in the event the Write function is
+			// expanded to have more preconditions.
+			err := Write("", "", test.Runtime, "", nil)
+			if test.Expected != nil && err == nil {
+				t.Fatalf("expected runtime %v to yield a detection error", test.Runtime)
+			}
+			if test.Expected != nil && !errors.As(err, test.Expected) {
+				t.Fatalf("did not receive expected error type for %v runtime.", test.Runtime)
+			}
+			t.Logf("ok: %v", err)
+		})
+	}
+}
+
+// TestWrite ensures that the Write method writes Scaffolding to the given
+// destination.  This is a failry shallow test.  See the Scaffolding and
+// Detector tests for more depth.
+func TestWrite(t *testing.T) {
+	// The filesystem containing scaffolding is expected to conform to the
+	// structure:
+	// /[language]/scaffolding/["instanced"|"static"]-[invocation]
```

Comment:
TODO: we will need to update existing language pack documentation with this kind of information